app/models/voucher.rb

```diff
@@ -10,7 +12,11 @@ class Voucher < ApplicationRecord
            class_name: 'Spree::Adjustment',
            dependent: :nullify
 
-  validates :code, presence: true, uniqueness: { scope: :enterprise_id }
+  before_validation :strip_code
+
+  validates :code, format: { without: INVALID_CODE_REGEX },
+                   length: { maximum: STRING_COLUMN_LIMIT },
```

Comment:
What do you think about using the limit of the database itself instead of repeating it in our own constant?
```suggestion
                   length: { maximum: columns_hash["code"].limit },
```

Apparently, there's a gem for it but it looks out of date:

* https://github.com/rubiety/validates_lengths_from_database

Reply:
That seems cool, but whether it's an activerecord validation error, or a database error, I guess the impact would be similar.

Even better if we can validate lengths on the frontend input. I once looked for a gem to add this automatically to the form builder but didn't find an answer.. 

But in any case, I think it would be better to worry about this once we deal with batch-creation.

Reply:
I have removed this https://github.com/openfoodfoundation/openfoodnetwork/pull/11259/commits/d026438edcd7f0211f1528dcef0318176b414fa1 for now seen as though it might be changing soon. Thanks for the review, let me know if you want me to rebase things.